src/main/java/org/springframework/integration/aws/outbound/SnsMessageHandler.java

```diff
@@ -113,6 +120,21 @@ public void setSubjectExpression(Expression subjectExpression) {
 		this.subjectExpression = subjectExpression;
 	}
 
+	public void setMessageGroupId(String messageGroupId) {
+	    Assert.hasText(messageGroupId, "messageGroupId must not be empty.");
+	    this.messageGroupIdExpression = new LiteralExpression(messageGroupId);
+	}
+
+	public void setMessageGroupIdExpression(Expression messageGroupIdExpression) {
+	    Assert.notNull(messageGroupIdExpression, "messageGroupIdExpression must not be null.");
+	    this.messageGroupIdExpression = messageGroupIdExpression;
+	}
+
+	public void setMessageDeduplicationIdExpression(Expression messageDeduplicationIdExpression) {
```

Comment:
Would you mind adding some simple Javadocs for these new methods an also mention `@since 2.5.3`?